king_phisher/client/dialogs/configuration.py

```diff
@@ -274,6 +291,27 @@ def interact(self):
 		self.dialog.destroy()
 		return response
 
+	def save_proxy_settings(self):
+		proxy_url = urllib.parse.urlparse(self.gtk_builder_get('entry_proxy_url').get_text().strip())
+		proxy_username = self.gtk_builder_get('entry_proxy_username').get_text().strip()
+		proxy_password = self.gtk_builder_get('entry_proxy_password').get_text().strip()
+		if proxy_url and not (proxy_url.hostname and proxy_url.scheme):
+			gui_utilities.show_dialog_warning('Invalid Proxy Settings', self.parent, 'The proxy url you have submitted is not valid.')
+			return
+		try:
+			proxy_url.port
+		except:
```

Comment:
This should except the specific exception class that should be raised, in this case that would be a `ValueError`.